.husky/pre-push

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+# Check if README.md is being pushed
+if git diff --name-only HEAD @{push} | grep -q "README.md"; then
+
+  # Get the diff content, removing the meta-headers
+  DIFF_CONTENT=$(git diff HEAD @{push} README.md | sed 1,4d)
+
+  # Look for added lines (+) outside the TOC section
+  DIFF_CONTENT_EXCLUDING_TOC=$(echo "$DIFF_CONTENT" | sed '/<!-- START doctoc/,/<!-- END doctoc/d' | grep '^\+' || true)
+
+  if [ -n "$DIFF_CONTENT_EXCLUDING_TOC" ]; then
+    echo "Changes detected outside TOC section. Running doctoc..."
+    npm run doctoc-readme
+    git add README.md
+    git commit --amend --no-edit
```

Comment:
This will automatically amend the commit that is about to be pushed to remote. It's seamless, so as a contributor you might not even realize your final commit has been changed before it was pushed. That said, it lowkey scares me so we could make it a manual process instead.

Reply:
Update: @esezen suggested maybe adding a new commit instead of amending the existing one, but it turns out that doesn't play well cc: https://stackoverflow.com/questions/21334493/git-commit-in-pre-push-hook